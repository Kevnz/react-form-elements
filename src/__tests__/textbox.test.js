import React from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import { render, fireEvent, cleanup } from '@testing-library/react'
import ReactDOMServer from 'react-dom/server'
import TextBox from '../textbox'

expect.extend(toHaveNoViolations)

describe('TextBox component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', async () => {
      const { getByText, container } = render(
        <TextBox label="My Label" initialValue="" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Label')
      expect(label.textContent).toBe('My Label')
    })
    it('should not violate', async () => {
      const html = ReactDOMServer.renderToString(
        <TextBox label="My Label" initialValue="" />
      )
      const results = await axe(html)
      expect(results).toHaveNoViolations()
    })
    it('should have a placeholder when passed', () => {
      const { getByText, container } = render(
        <TextBox
          label="My Label"
          initialValue=""
          placeholder="textbox placeholder"
        />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Label')
      expect(label.textContent).toBe('My Label')
    })
    it('should have a no label when passed an empty string as label', () => {
      const { container } = render(
        <TextBox label="" initialValue="" placeholder="textbox placeholder" />
      )
      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { getByLabelText, container } = render(
        <TextBox label="My Label" initialValue="My Input Value" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Label')
      expect(input.value).toBe('My Input Value')
    })
    it('should update a value when changed', () => {
      const { getByLabelText, container } = render(
        <TextBox label="My Label" initialValue="My Input Value" />
      )
      const input = getByLabelText('My Label')

      fireEvent.change(input, { target: { value: 'Updated Value' } })
      expect(input.value).toBe('Updated Value')
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
