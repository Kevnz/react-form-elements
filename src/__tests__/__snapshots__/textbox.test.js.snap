// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Label
  </label>
  <input
    id="id-1-1"
    name="TextBox"
    type="text"
    value=""
  />
</div>
`;

exports[`TextBox component The Rendered HTML should have a no label when passed an empty string as label 1`] = `
<div
  class="rfe-form__row"
>
  <input
    id="id-4-1"
    name="TextBox"
    placeholder="textbox placeholder"
    type="text"
    value=""
  />
</div>
`;

exports[`TextBox component The Rendered HTML should have a placeholder when passed 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-3-1"
  >
    My Label
  </label>
  <input
    id="id-3-1"
    name="TextBox"
    placeholder="textbox placeholder"
    type="text"
    value=""
  />
</div>
`;

exports[`TextBox component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-5-1"
  >
    My Label
  </label>
  <input
    id="id-5-1"
    name="TextBox"
    type="text"
    value="My Input Value"
  />
</div>
`;

exports[`TextBox component handling input should update a value when changed 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-7-1"
  >
    My Label
  </label>
  <input
    id="id-7-1"
    name="TextBox"
    type="text"
    value="Updated Value"
  />
</div>
`;
